lib/spack/spack/util/compression.py

```diff
@@ -285,8 +305,19 @@ def decompressor_for(path, ext):
     if re.match(r"xz", ext):
         return _lzma_decomp
 
-    if ("xz" in ext or "Z" in ext) and is_windows:
-        return _7zip
+    # Windows tar does not come with lzma or .Z support
+    # so we catch tarballs with those compressions and
+    # handle with 7zip if no native tar/lzma support
+    if is_windows:
+        if "xz" in ext:
+            # we can rely on Python's native lzma/tar support
+            if txz_support():
+                return _untar
+            else:
+                # we only have guarantee of system support
+                return _7zip
+        if "Z" in ext:
+            return _7zip
```

Comment:
If I understand, this isn't strictly needed: it looks like the change is to return `_untar` when possible in order to rely on native support when possible. Given that it falls back on `_7zip` - which is was always using before - I assume this particular change can be reverted. If that's true, I recommend rewording:

```
# Windows tar does not come with lzma or .Z support
# so we catch tarballs with those compressions and
# handle with 7zip if no native tar/lzma support
```

as

```
Use native decompression for .xz-compressed archives when possible.
```

Reply:
Like the other compression change, this is best handled in #31589 as that PR addresses this issue more directly.